components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -209,12 +261,115 @@ func (sb *sensorBase) isPolling() bool {
 func (sb *sensorBase) MoveStraight(
 	ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{},
 ) error {
-	sb.stopLoop()
 	ctx, done := sb.opMgr.New(ctx)
 	defer done()
 	sb.setPolling(false)
+	straightTimeEst := time.Duration(int(time.Second) * int(math.Abs(float64(distanceMm)/mmPerSec)))
+	startTime := time.Now()
+	timeOut := 5 * straightTimeEst
+	if timeOut < 10*time.Second {
+		timeOut = 10 * time.Second
+	}
+
+	if sb.position == nil || len(sb.conf.ControlParameters) == 0 {
+		sb.logger.CWarn(ctx, "Position not given, using base MoveStraight")
+		sb.stopLoop()
+		return sb.controlledBase.MoveStraight(ctx, distanceMm, mmPerSec, extra)
+	}
+
+	if sb.headingFunc == nil {
+		sb.logger.CDebug(ctx, "No movement sensors found for heading")
+	}
+
+	// make sure the control loop is enabled
+	if sb.loop == nil {
+		if err := sb.startControlLoop(); err != nil {
+			return err
+		}
+	}
+
+	// initialize relevant parameters for moving straight
+	slowDownDist := calcSlowDownDist(distanceMm)
+	var initPos *geo.Point
+	var err error
+	if sb.position != nil {
+		initPos, _, err = sb.position.Position(ctx, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	ticker := time.NewTicker(time.Duration(1000./sb.controlLoopConfig.Frequency) * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			var errDist float64
+
+			if sb.position != nil {
```

Comment:
checking if position is set here for when I eventually add support for velocity based move straight calls